src/main/java/com/ivan/salesapp/services/ProductService.java

```diff
@@ -29,21 +33,54 @@ public ProductServiceModel addProduct(ProductServiceModel productServiceModel) {
 
     @Override
     public List<ProductServiceModel> findAllProducts() {
-        return null;
+        return this.productRepository.findAll().stream()
+                .map(p -> this.modelMapper
+                        .map(p, ProductServiceModel.class)).collect(Collectors.toList());
     }
 
     @Override
     public ProductServiceModel findProductById(String id) {
-        return null;
+        Product product = this.productRepository.findById(id)
+                .orElseThrow(() -> new IllegalArgumentException("Incorrect product id!"));
+
+        return this.modelMapper.map(product, ProductServiceModel.class);
     }
 
     @Override
-    public ProductServiceModel editProduct(String id, ProductServiceModel productServiceModel) {
-        return null;
+    public ProductServiceModel editProduct(String id, ProductServiceModel productServiceModel, List<String> categories) {
+        Product product = this.productRepository.findById(id)
+                .orElseThrow(() -> new IllegalArgumentException("Incorrect id!"));
+
+        product.setName(productServiceModel.getName());
+        product.setDescription(productServiceModel.getDescription());
+        product.setPrice(productServiceModel.getPrice());
+
+        product.setCategories(
+                this.iCategoryService.findAllCategories()
+                        .stream()
+                        .filter(c -> categories.contains(c.getId()))
+                        .map(c -> this.modelMapper.map(c, Category.class))
+                        .collect(Collectors.toList())
+        );
+
+
+        return this.modelMapper.map(this.productRepository.saveAndFlush(product), ProductServiceModel.class);
     }
 
     @Override
     public void deleteProduct(String id) {
+        Product product = this.productRepository.findById(id)
+                .orElseThrow(() -> new IllegalArgumentException("Incorrect id!"));
+
+        this.productRepository.delete(product);
+    }
 
+    @Override
+    public List<ProductServiceModel> findAllByCategory(String category) {
+        return this.productRepository.findAll()
+                .stream()
+                .filter(product -> product.getCategories().stream().anyMatch(categoryStream -> categoryStream.getName().equals(category)))
```

Comment:
this one is too complex product -> product.getCategories().stream().anyMatch(categoryStream -> categoryStream.getName().equals(category)

you can extract it to a new method and just to call filter(someMethod(..))